### ISSUE TITLE
AttributeError when specifying "Rescaled" regression learners and grid_search=false

### ISSUE THREAD

**Issue by bwriordan** [EducationalTestingService/skll#304]
Occurs when adding "Rescaled" to any regressor name and grid_search=false. Runs with grid_search=true. Observed this for tasks `train` and `cross_validate`.

```
Traceback (most recent call last):
  File "/Users/briordan/anaconda/bin/run_experiment", line 6, in <module>
    sys.exit(main())
  File "/Users/briordan/anaconda/lib/python3.4/site-packages/skll/utilities/run_experiment.py", line 108, in main
    ablation=ablation, resume=args.resume)
  File "/Users/briordan/anaconda/lib/python3.4/site-packages/skll/experiments.py", line 848, in run_configuration
    _classify_featureset(job_args)
  File "/Users/briordan/anaconda/lib/python3.4/site-packages/skll/experiments.py", line 483, in _classify_featureset
    iteritems(learner.model.get_params()))
AttributeError: 'NoneType' object has no attribute 'get_params'
```

**Comment by aoifecahill** [comment]
Is this a general problem for all datasets, or is it also particular to the dataset?

**Comment by bwriordan** [comment]
This is a problem for the dataset I'm working with. You mentioned a problem on another dataset. I haven't tried others.

**Comment by aoifecahill** [comment]
Right. Something to keep in mind when looking into it. It could just be some strange data, or it might be a more widespread issue that for some reason we have not come across before.

**Comment by bwriordan** [comment]
I replicated this basic error on the boston dataset in the examples folder. Below are my config file and the traceback.
```
[General]
experiment_name = Example_CV
task = cross_validate

[Input]
# this could also be an absolute path instead (and must be if you're not running things in local mode)
train_directory = train
featuresets = [["example_boston_features"]]
# there is only set of features to try with one feature file in it here.
featureset_names = ["example_boston"]
learners = ["RescaledSVR"]
#learners = ["RescaledLinearRegression"]
#learners = ["RandomForestRegressor", "SVR", "LinearRegression"]
suffix = .jsonlines

[Tuning]
grid_search = false
objective = pearson

[Output]
# again, these can be absolute paths
results = output
log = output
predictions = output
```

```
Traceback (most recent call last):
  File "/Users/briordan/anaconda/envs/sklltest/bin/run_experiment", line 9, in <module>
    load_entry_point('skll', 'console_scripts', 'run_experiment')()
  File "/Users/briordan/scratch/sklltest/skll/skll/utilities/run_experiment.py", line 108, in main
    ablation=ablation, resume=args.resume)
  File "/Users/briordan/scratch/sklltest/skll/skll/experiments.py", line 876, in run_configuration
    _classify_featureset(job_args)
  File "/Users/briordan/scratch/sklltest/skll/skll/experiments.py", line 465, in _classify_featureset
    grid_jobs=grid_search_jobs, save_cv_folds=save_cv_folds)
  File "/Users/briordan/scratch/sklltest/skll/skll/learner.py", line 1512, in cross_validate
    grid_objective=grid_objective))
  File "/Users/briordan/scratch/sklltest/skll/skll/learner.py", line 1124, in evaluate
    append=append)
  File "/Users/briordan/scratch/sklltest/skll/skll/learner.py", line 1296, in predict
    else self._model.predict(xtest))
AttributeError: 'NoneType' object has no attribute 'predict'
```

**Comment by desilinguist** [comment]
Yes, this seems to be a general bug. The root cause is that `learner.model` is `None` even after it has been fit on the data. This can be confirmed via the API using the "boston" example data:

```python
from skll import Learner
from skll.data import NDJReader

reader = NDJReader.for_path('examples/boston/train/example_boston_features.jsonlines')
examples = reader.read()

L1 = Learner('SVR')
L2 = Learner('RescaledSVR')

L1.train(examples, grid_search=False)
L2.train(examples, grid_search=False)

print(L1.model)
print(L2.model)
```

This outputs the following:

```
SVR(C=1.0, cache_size=1000, coef0=0.0, degree=3, epsilon=0.1, gamma='auto',
  kernel='rbf', max_iter=-1, shrinking=True, tol=0.001, verbose=False)

None
```

I believe this happens because the `fit()` method in the `rescaled()` decorator does not return `self` as it is supposed to. This is the norm in scikit-learn. However, when `grid_search` is `True`, the `GridSearchCV` gets around this issue by explicitly setting the `best_estimator_` to the right class in sklearn-land with the best parameters.

I am going to make a PR.

**Comment by desilinguist** [comment]
Addressed by #306.
